src/libcollections/linked_list.rs

```diff
@@ -609,12 +635,21 @@ impl<T> LinkedList<T> {
             length: len - at
         };
 
-        // Swap split_node.next with list_head (which is None), nulling out split_node.next,
-        // as it is the new tail.
-        mem::swap(&mut split_node.resolve().unwrap().next, &mut splitted_list.list_head);
```

Comment:
Actually, this could have just been done in the ctor of splitted_list all along, right?

```
let mut splitted_list = LinkedList {
    list_head: split_node.resolve().unwrap().next.take()
    list_tail: self.list_tail,
    length: len - at,
}
```